controllers/reconcilers/kserve_isvc_gateway_reconciler.go

```diff
@@ -0,0 +1,279 @@
+		errs = append(errs, err)
+	}
+
+	log.V(1).Info(fmt.Sprintf("Deleting the Server(%s) from KServe local gateway in the %s namespace", fmt.Sprintf("%s-%s", "https", isvc.Name), meshNamespace))
+	if err := r.removeServerFromGateway(ctx, log, fmt.Sprintf("%s-%s", "https", isvc.Name)); err != nil {
+		log.V(1).Error(err, fmt.Sprintf("Failed to remove the Server(%s) from KServe local gateway in the %s namespace", fmt.Sprintf("%s-%s", "https", isvc.Name), meshNamespace))
+		errs = append(errs, err)
+	}
+
+	if len(errs) > 0 {
+		return fmt.Errorf("multiple errors: %v", errs)
+	}
+
+	return nil
+}
+
+func (r *KserveGatewayReconciler) Cleanup(ctx context.Context, log logr.Logger, isvcName string) error {
+	// NOOP - Resources should not be deleted until the kserve component is uninstalled.
+	return nil
+}
+
+func (r *KserveGatewayReconciler) processDelta(ctx context.Context, log logr.Logger, desiredGateway *istioclientv1beta1.Gateway, existingGateway *istioclientv1beta1.Gateway) (err error) {
+	comparator := comparators.GetGatewayComparator()
+	delta := r.deltaProcessor.ComputeDelta(comparator, desiredGateway, existingGateway)
+
+	if delta.IsUpdated() {
+		log.V(1).Info("Delta found", "update", desiredGateway.GetName())
+		gw := existingGateway.DeepCopy()
+		gw.Spec.Servers = append(existingGateway.Spec.Servers, desiredGateway.Spec.Servers[0])
+
+		if err = r.gatewayHandler.Update(ctx, gw); err != nil {
+			log.V(1).Error(err, fmt.Sprintf("Failed to add the Server(%s) from KServe local gateway in the istio-system namespace", desiredGateway.Spec.Servers[0].Port.Name))
+			return err
+		}
+		return nil
+	}
+	return nil
+}
+
+func (r *KserveGatewayReconciler) removeServerFromGateway(ctx context.Context, log logr.Logger, serverToRemove string) error {
+	gateway, err := r.gatewayHandler.Get(ctx, types.NamespacedName{Name: constants.KServeGatewayName, Namespace: meshNamespace})
+	if err != nil {
+		log.V(1).Error(err, "Failed to retrieve KServe local gateway in istio-system namespace")
+		return err
+	}
+
+	newServers := []*istiov1beta1.Server{}
+	for _, server := range gateway.Spec.Servers {
+		if server.Port.Name != serverToRemove {
+			newServers = append(newServers, server)
+		}
+	}
+
+	gateway.Spec.Servers = newServers
+	if err := r.gatewayHandler.Update(ctx, gateway); err != nil {
+		log.V(1).Error(err, "Failed to update KServe local gateway in istio-system namespace")
+		return err
+	}
+
+	return nil
+}
+
+func (r *KserveGatewayReconciler) copyServingCertSecretFromIsvcNamespace(ctx context.Context, sourceSecret *corev1.Secret, preDestSecret *corev1.Secret) error {
+	destinationSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-%s", sourceSecret.Name, sourceSecret.Namespace),
+			Namespace: meshNamespace,
+			Labels: map[string]string{
+				"opendatahub.io/managed": "true",
+			},
```

Comment:
There is an ADR covering recommended labels, although not merged yet. Additionally to this one, not sure if you would like to follow the ADR and add the other suggested labels: https://github.com/opendatahub-io/architecture-decision-records/pull/26/files#diff-87d90e8a91b0f96b79615d8b6d9887902a66d2a4ab38415757fc7945b94068a0R94.

Example here: https://github.com/opendatahub-io/odh-model-controller/blob/86cc36042347e1817f7263aacc1ceb818468e85b/controllers/reconcilers/kserve_istio_smm_reconciler.go#L73-L77.

This is just a suggestion. Not a blocker...

Reply:
good point!. i added the labels.